moped-editor/src/views/projects/projectView/ProjectSummary.js

```diff
@@ -5,21 +5,99 @@ import ProjectSummaryTable from "./ProjectSummaryTable";
 import ProjectSummaryMap from "./ProjectSummaryMap";
 import { createFeatureCollectionFromProjectFeatures } from "../../../utils/mapHelpers";
 
-import { Grid, CardContent, CircularProgress } from "@material-ui/core";
+import {
+  Grid,
+  CardContent,
+  CircularProgress,
+  TextField,
+  Icon,
+  Box,
+} from "@material-ui/core";
 import ApolloErrorHandler from "../../../components/ApolloErrorHandler";
 
+import ControlPointIcon from "@material-ui/icons/ControlPoint";
+
+import {
+  PROJECT_SUMMARY_STATUS_UPDATE_INSERT,
+  PROJECT_SUMMARY_STATUS_UPDATE_UPDATE,
+} from "../../../queries/project";
+
 /*
   Error Handler and Fallback Component
 */
 import ProjectSummaryMapFallback from "./ProjectSummaryMapFallback";
 import { ErrorBoundary } from "react-error-boundary";
+import makeStyles from "@material-ui/core/styles/makeStyles";
+import { useMutation } from "@apollo/client";
+import { getSessionDatabaseData } from "../../../auth/user";
+
+const useStyles = makeStyles(theme => ({
+  fieldGridItem: {
+    margin: theme.spacing(2),
+  },
+  editIcon: {
+    cursor: "pointer",
+    margin: "28px 0 0 8px",
+    fontSize: "20px",
+  },
+  editIconConfirm: {
+    cursor: "pointer",
+    margin: "28px 0 0 8px",
+    fontSize: "24px",
+  },
+}));
 
+/**
+ * Project Summary Component
+ * @param {boolean} loading - True if it is loading
+ * @param {Object} error - Error content if provided
+ * @param {Object} data - THe query data
+ * @param {function} refetch - A function to reload the data
+ * @return {JSX.Element}
+ * @constructor
+ */
 const ProjectSummary = ({ loading, error, data, refetch }) => {
   const { projectId } = useParams();
+  const classes = useStyles();
+  const userSessionData = getSessionDatabaseData();
 
   const [makeSureRefresh, setMakeSureRefresh] = useState(false);
   const [mapError, setMapError] = useState(false);
 
+  const [updateProjectStatusUpdateInsert] = useMutation(
+    PROJECT_SUMMARY_STATUS_UPDATE_INSERT
+  );
+
+  const [updateProjectStatusUpdateUpdate] = useMutation(
+    PROJECT_SUMMARY_STATUS_UPDATE_UPDATE
+  );
+
+  /**
+   * Retrieve the last note made as a status update (which are ordered by date)
+   * @param {String} fieldName - The name of the field to retrieve in the moped_proj_note object
+   * @return {string|null}
+   */
+  const getStatusUpdate = (fieldName = "project_note") => {
+    const lastItem =
+      (data?.moped_project[0]?.moped_proj_notes?.length ?? 0) - 1;
+    if (lastItem >= 0) {
+      // Get the data from the note
+      const note = data
+        ? data?.moped_project[0].moped_proj_notes[lastItem][fieldName] ?? ""
+        : null;
+      // Remove any HTML tags
+      return note ? String(note).replace(/(<([^>]+)>)/gi, "") : null;
+    }
+    return null;
+  };
```

Comment:
An important function, basically retrieves the last comment of type 2 and makes that the last status update.